errors/errctx.go

```diff
@@ -48,3 +49,60 @@ func Wrapf(ctx context.Context, err error, format string, args ...interface{}) e
 func Errorf(ctx context.Context, format string, args ...interface{}) error {
 	return ErrCtx{ctx: ctx, err: errors.Errorf(format, args...)}
 }
+
+// GetDeepestContext unwrap all wrapped errors from err to get the deepest context
+// from ErrCtx errors. If there is no wrapped ErrCtx, GetDeepestContext returns nil.
+func GetDeepestContext(err error) context.Context {
+	log := logger.Default()
+
+	var lastCtx context.Context
+
+	type causer interface {
+		Cause() error
+	}
+
+	// Unwrap each error to get the deepest context
+	for err != nil {
+		log.Debug("Unwrapping error type '%T'\n", err)
```

Comment:
I'm really not super fan of adding custom debug logging (impacted by environment) in a lib like t his, tests should ensure it works. These `log.Debug` are to be removed for me.